src/software/proto/message_translation/protobuf_primitive_visitor.cpp

```diff
@@ -0,0 +1,161 @@
+#include "software/proto/message_translation/protobuf_primitive_visitor.h"
+
+#include "shared/constants.h"
+#include "shared/proto/radio_primitive.pb.h"
+#include "software/primitive/all_primitives.h"
+
+std::unique_ptr<RadioPrimitiveMsg> ProtobufPrimitiveVisitor::getRadioPrimitiveMsg()
+{
+    // If we've never visited a primitive (and so have never populated the
+    // `prim_msg_ptr`) then throw an exception
+    if (!prim_msg_ptr)
+    {
+        std::string err_msg = std::string(typeid(this).name()) + ": " + __func__ +
+                              " called without ever having visited anything";
+        throw std::runtime_error(err_msg);
+    }
```

Comment:
We can avoid this by taking a primitive as an argument. Take a look at how Navigator is done.

Reply:
fixed